src/routes/uniswap.route.js

```diff
@@ -0,0 +1,308 @@
+import BigNumber from 'bignumber.js';
+import { ethers } from 'ethers';
+import express from 'express';
+
+import { getParamData, latency, reportConnectionError, statusMessages } from '../services/utils';
+import Uniswap from '../services/uniswap';
+import Ethereum from '../services/eth';
+
+require('dotenv').config()
+const debug = require('debug')('router')
+
+const router = express.Router()
+const uniswap = new Uniswap(process.env.UNISWAP_NETWORK)
+const eth = new Ethereum(process.env.UNISWAP_NETWORK)
+
+const denomMultiplier = 1e18
+const swapMoreThanMaxPriceError = 'Swap price exceeds maxPrice'
+const swapLessThanMaxPriceError = 'Swap price lower than maxPrice'
+
+router.use((req, res, next) => {
+  const cert = req.connection.getPeerCertificate()
+  if (req.client.authorized) {
+    next()
+  } else if (cert.subject) {
+    res.status(403).send({ error: statusMessages.ssl_cert_invalid })
+  } else {
+    res.status(401).send({ error: statusMessages.ssl_cert_required })
+  }
+})
+
+router.post('/', async (req, res) => {
+  /*
+    POST /
+  */
+  res.status(200).json({
+    network: uniswap.network,
+    provider: uniswap.provider.connection.url,
+    exchangeProxy: uniswap.exchangeProxy,
+    subgraphUrl: process.env.REACT_APP_SUBGRAPH_URL, // would most likely replace with uniswap's when working on the client connector
+    connection: true,
+    timestamp: Date.now(),
+  })
+})
+
+router.post('/sell-price', async (req, res) => {
+  /*
+    POST: /sell-price
+      x-www-form-urlencoded: {
+        "quote":"0x....."
+        "base":"0x....."
+        "amount":0.1
+      }
+  */
+  const initTime = Date.now()
+  // params: base (required), quote (required), amount (required)
+  const paramData = getParamData(req.body)
+  const baseTokenAddress = paramData.base
+  const quoteTokenAddress = paramData.quote
+
+  try {
+    // fetch the optimal pool mix from uniswap
+    const route = await uniswap.fetch_route(
+      baseTokenAddress,     // tokenIn is base asset
+      quoteTokenAddress,    // tokenOut is quote asset
+    )
+
+    if (route != null) {
+      res.status(200).json({
+        network: uniswap.network,
+        timestamp: initTime,
+        latency: latency(initTime, Date.now()),
+        base: baseTokenAddress,
+        quote: quoteTokenAddress,
+        amount: parseFloat(paramData.amount),
+        expectedOut: parseFloat(paramData.amount * route.midPrice.toSignificant(8)),
+        price: route.midPrice.toSignificant(8),
```

Comment:
I don't think the mid price is a good way to fetch the price, since slippage in Uniswap pools is based on 1) order amount and 2) size of reserves in the pool. So for large orders or thin pools, the buy and sell prices can differ from mid greatly.

Let's discuss during our 1:1 tomorrow how to fetch buy and sell prices.